test/samples.jl

```diff
@@ -120,6 +120,83 @@
     end
 end
 
+@testset "`decode` promote" begin
+    data = [1,2,3]
+    @test decode(1, 0, data) === data
+
+    data = [1,2,3]
+    result = decode(1.0, 0.0, data)
+    @test eltype(result) === Float64
+    @test result !== data
+    @test result == data
+
+    @testset "normal signals" begin
+        # Signal must use a non-one resolution and a non-zero offset
+        info = SamplesInfoV2(; sensor_type="eeg",
+                             channels=["a", "b"],
+                             sample_unit="unit",
+                             sample_resolution_in_unit=1,
+                             sample_offset_in_unit=1,
+                             sample_type=Int16,
+                             sample_rate=200)
+        samples = Samples(rand(-Int16(20):Int16(20), 2, 5), info, false)
+
+        decoded = decode(samples)
+        @test eltype(decoded.data) === Int16
+        @test decoded.data === samples.data
+
+        decoded = decode(samples, Float32)
+        @test eltype(decoded.data) === Int16
```

Comment:
We may want to have it such that `decode` automatically converts the element type on data which doesn't match the specified type. I think I've done enough here so I'll leave that for a future contributor.

Reply:
This would be the remaining work to do on #141

Reply:
Seems like the remaining concerns are just about this portion so I'll get this done after all